File: src/main/java/controller/SnakeController.java
package controller;

import model.Arena;
import model.ModifiableArena;
import view.ArenaView;

import javax.swing.*;
import java.awt.*;

public class SnakeController {

    private static final int WIDTH = 550;
    private static final int HEIGHT = 570;
    private static final String TITLE = "Snake";
    private static final int DELAY_BETWEEN_STEPS = 1;

    private static final Dimension FRAME_SIZE = new Dimension(WIDTH, HEIGHT);

    private final JFrame frame = new JFrame(TITLE);
    private final Timer timer = new Timer(DELAY_BETWEEN_STEPS, new TimerAction(this));

    private ArenaView arenaView;

    private final ModifiableArena arena;

    public SnakeController(ModifiableArena arena) {
        this.arena = arena;
        initView(arena);
        frame.setVisible(true);
    }

    public void start() {
        timer.start();
    }

    public void step() {
        arena.move();
        arenaView.repaint();
    }

    public void stop() {
        timer.stop();
        arenaView.repaint();
    }

    private void initView(Arena arena) {
        this.arenaView = new ArenaView(arena);
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.add(arenaView);
        frame.setSize(FRAME_SIZE);
        frame.setLocationRelativeTo(null);
    }

}

File: src/main/kotlin/co/electric/snake/controller/SnakeController.kt
package co.electric.snake.controller

import co.electric.snake.model.Arena
import co.electric.snake.model.ModifiableArena
import co.electric.snake.view.ArenaView
import java.awt.Dimension
import javax.swing.JFrame
import javax.swing.Timer

class SnakeController(private val arena: ModifiableArena) {

    companion object {
        private const val WIDTH = 550
        private const val HEIGHT = 570
        private const val TITLE = "Snake"
        private const val DELAY_BETWEEN_STEPS = 1

        private val FRAME_SIZE = Dimension(WIDTH, HEIGHT)
    }

    private val frame = JFrame(TITLE)
    private val timer = Timer(DELAY_BETWEEN_STEPS, TimerAction(this))

    private lateinit var arenaView: ArenaView

    init {
        initArenaView(arena)
        frame.isVisible = true
    }

    fun start() {
        timer.start()
    }

    fun stop() {
        timer.stop()
        arenaView.repaint()
    }

    fun step() {
        arena.move()
        arenaView.repaint()
    }

    private fun initArenaView(arena: Arena) {
        arenaView = ArenaView(arena)
        frame.defaultCloseOperation = JFrame.EXIT_ON_CLOSE
        frame.add(arenaView)
        frame.size = FRAME_SIZE
        frame.setLocationRelativeTo(null)
    }

}
